Sources/PackageRegistry/SourceArchiver.swift

```diff
@@ -1,72 +0,0 @@
-/*
- This source file is part of the Swift.org open source project
-
- Copyright (c) 2021 Apple Inc. and the Swift project authors
- Licensed under Apache License v2.0 with Runtime Library Exception
-
- See http://swift.org/LICENSE.txt for license information
- See http://swift.org/CONTRIBUTORS.txt for Swift project authors
- */
-
-import Basics
-import TSCBasic
-import TSCUtility
-
-import Dispatch
-
-/// An `Archiver` that handles source archives.
-///
-/// Source archives created with `swift package archive-source`
-/// have a top-level directory prefix (for example, `LinkedList-1.1.1/`).
-/// Unfortunately, the `unzip` command used by `ZipArchiver` doesn't have an option for
-/// ignoring this top-level prefix.
-/// Rather than performing additional (possibly unsafe) file system operations,
-/// this `Archiver` delegates to `tar`, which has a built-in `--strip-components=` option.
-struct SourceArchiver: Archiver {
-    public var supportedExtensions: Set<String> { ["zip"] }
-
-    /// The file-system implementation used for various file-system operations and checks.
-    private let fileSystem: FileSystem
-
-    /// Creates a `SourceArchiver`.
-    ///
-    /// - Parameters:
-    ///   - fileSystem: The file-system to used by the `SourceArchiver`.
-    public init(fileSystem: FileSystem) {
-        self.fileSystem = fileSystem
-    }
-
-    public func extract(
-        from archivePath: AbsolutePath,
-        to destinationPath: AbsolutePath,
-        completion: @escaping (Result<Void, Error>) -> Void
-    ) {
-        guard fileSystem.exists(archivePath) else {
-            completion(.failure(FileSystemError(.noEntry, archivePath)))
-            return
-        }
-
-        guard fileSystem.isDirectory(destinationPath) else {
-            completion(.failure(FileSystemError(.notDirectory, destinationPath)))
-            return
-        }
-
-        // TODO: consider calling `libarchive` or some other library directly instead of spawning a process
-        DispatchQueue.sharedConcurrent.async {
-            do {
-                let result = try Process.popen(args: "bsdtar",
```

Comment:
`bsdtar` is a new system dependency and the reason it was used as explained above (see code comment) was for stripping the top level directory. this PR basically rolls back that and uses `unzip` which SwiftPM already relies on (and as such is already a system dependency) and handles stripping programmatically after the unzipping